src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -565,7 +565,7 @@ private static size_t CurlReceiveHeadersCallback(IntPtr buffer, size_t size, siz
                             if (colonIndex > 0)
                             {
                                 string headerName = responseHeader.Substring(0, colonIndex);
-                                string headerValue = responseHeader.Substring(colonIndex + 1);
+                                string headerValue = responseHeader.Substring(colonIndex + 1).Trim();
```

Comment:
@kapilash, if you'd like to incorporate the fix into your change, that's fine, too, and you can close this. Either way.